src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1119,36 +1119,77 @@ PeerImp::onMessage(std::shared_ptr <protocol::TMShardInfo> const& m)
     // Parse the shard indexes received in the shard info
```

Comment:
Maybe amend this comment to explain that we're (1) parsing indices from a comma-separated string of indices (e.g. 123) and index ranges (e.g. 456-789), and (2) raising an error if any of them are out of some bounds that we assume.